src/renderer/components/+extensions/extensions.tsx

```diff
@@ -35,458 +57,457 @@ interface InstallRequestPreloaded extends InstallRequest {
     );
-
-    return validatedRequests;
   }
 
-  async requestInstall(init: InstallRequest | InstallRequest[]) {
-    const requests = Array.isArray(init) ? init : [init];
-    const preloadedRequests = await this.preloadExtensions(requests);
-    const validatedRequests = await this.createTempFilesAndValidate(preloadedRequests);
-
-    // If there are no requests for installing, reset startingInstall state
-    if (validatedRequests.length === 0) {
-      this.startingInstall = false;
-    }
-
-    for (const install of validatedRequests) {
-      const { name, version, description } = install.manifest;
-      const extensionFolder = this.getExtensionDestFolder(name);
-      const folderExists = await fse.pathExists(extensionFolder);
-
-      if (!folderExists) {
-        // auto-install extension if not yet exists
-        this.unpackExtension(install);
-      } else {
-        // If we show the confirmation dialog, we stop the install spinner until user clicks ok
-        // and the install continues
-        this.startingInstall = false;
-
-        // otherwise confirmation required (re-install / update)
-        const removeNotification = Notifications.info(
-          <div className="InstallingExtensionNotification flex gaps align-center">
-            <div className="flex column gaps">
-              <p>Install extension <b>{name}@{version}</b>?</p>
-              <p>Description: <em>{description}</em></p>
-              <div className="remove-folder-warning" onClick={() => shell.openPath(extensionFolder)}>
-                <b>Warning:</b> <code>{extensionFolder}</code> will be removed before installation.
-              </div>
-            </div>
-            <Button autoFocus label="Install" onClick={() => {
-              removeNotification();
-              this.unpackExtension(install);
-            }}/>
+  const extensionFolder = getExtensionDestFolder(name);
+  const folderExists = await fse.pathExists(extensionFolder);
+
+  if (!folderExists) {
+    // install extension if not yet exists
+    await unpackExtension(validatedRequest, dispose);
+  } else {
+    // otherwise confirmation required (re-install / update)
+    const removeNotification = Notifications.info(
+      <div className="InstallingExtensionNotification flex gaps align-center">
+        <div className="flex column gaps">
+          <p>Install extension <b>{name}@{version}</b>?</p>
+          <p>Description: <em>{description}</em></p>
+          <div className="remove-folder-warning" onClick={() => shell.openPath(extensionFolder)}>
+            <b>Warning:</b> <code>{extensionFolder}</code> will be removed before installation.
           </div>
-        );
+        </div>
+        <Button autoFocus label="Install" onClick={async () => {
+          removeNotification();
+
+          if (await uninstallExtension(validatedRequest.id, validatedRequest.manifest)) {
+            await unpackExtension(validatedRequest, dispose);
+          } else {
+            dispose();
```

Comment:
Does the `validatedRequest.tempFile` need to be removed at this point?

Reply:
Yes